presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/ParquetPredicateUtils.java

```diff
@@ -188,14 +190,37 @@ private static boolean isColumnPredicate(ColumnDescriptor columnDescriptor, Tupl
                 .anyMatch(columnName::equals);
     }
 
-    private static boolean isOnlyDictionaryEncodingPages(Set<Encoding> encodings)
+    @VisibleForTesting
+    @SuppressWarnings("deprecation")
+    static boolean isOnlyDictionaryEncodingPages(Set<Encoding> encodings)
     {
-        // more than 1 encodings for values
-        if (encodings.size() > PARQUET_DATA_TRIPLE) {
+        if (encodings.contains(PLAIN_DICTIONARY)) {
+            // if remove returned true, PLAIN_DICTIONARY was present, which means at
+            // least one page was dictionary encoded and 1.0 encodings are used
+            int expectedEncodings = 1; // for PLAIN_DICTIONARY
+
+            // RLE and BIT_PACKED are only used for repetition or definition levels
+            if (encodings.contains(RLE)) {
+                expectedEncodings += 1;
+            }
+
+            if (encodings.contains(BIT_PACKED)) {
+                expectedEncodings += 1;
+            }
+
+            if (encodings.size() > expectedEncodings) {
+                return false; // no encodings other than dictionary or rep/def levels
+            }
+
+            return true;
+
+        }
+        else {
+            // if PLAIN_DICTIONARY wasn't present, then either the column is not
+            // dictionary-encoded, or the 2.0 encoding, RLE_DICTIONARY, was used.
+            // for 2.0, this cannot determine whether a page fell back without
+            // page encoding stats
             return false;
         }
-        // definition level, repetition level never have dictionary encoding
-        // TODO: add PageEncodingStats in ColumnChunkMetaData
```

Comment:
I've updated it for the implementation that will be released in Parquet 1.9.0 and added it back.